app/controllers/option_sets_controller.rb

```diff
@@ -133,16 +133,14 @@ def prep_form_vars
 
   # creates/updates the option set
   def create_or_update
-    if @option_set.save
-      create_or_update_success
-    else
-      error = if @option_set.errors.attribute_names[0] == :name
-                :bad_request
-              else
-                :internal_server_error
-              end
-      render(json: "error", status: error)
-    end
+    @option_set.save!
+    raise StandardError if @option_set.errors?.attribute_names[0] == :name
+  rescue ActiveRecord::DeleteRestrictionError
+    render(json: "error", status: :conflict)
+  rescue StandardError
+    render(json: "error", status: :bad_request)
+  else
```

Comment:
maybe there is a better way to do this that doesn't have multiple render() commands?

Reply:
ooh this is looking so much better! nice!

you could write a separate render_errors method that takes the status as a param?

Reply:
Just attempted this! But I'm noticing many other render() calls in the controller as well...

Reply:
this is beautiful!!!! 

yeah, that's ok, it's more of a best practice.